### PR TITLE
store/gc_worker: Use UnsafeDestroyRange instead of DeleteRange in GC

### PR DESCRIPTION
**Pull request by MyonKeminta**
### What problem does this PR solve? 

This PR tries to solve the problem that the third step of GC, DeleteRanges, is too slow.

### What is changed and how it works?

* Add `GetAllStores` function to PdClient
* Add PdClient as a member of GCWorker.
* -[x] In gc_worker, the logic of DeleteRanges is changed. Previously it calls DeleteRanges on each region, now it calls DestroyRange on each TiKV.
* -[x] Check `delete_range_done` table and pick ranges deleted 24 hours again, and call DestroyRange again

The idea is detailed recorded in https://github.com/tikv/rfcs/pull/2

This PR should not be merged before https://github.com/tikv/tikv/pull/3560

### Check List 

Tests 

 - Manual test (Results can be found at https://github.com/tikv/rfcs/pull/2 , more tests are in progress)

Code changes

 - Has interface methods change
   - Constructor of GCWorker needs a PdClient now.

Side effects

 - Increased code complexity
 - Breaking backward compatibility

### CODE REVIEW

**Comment by jackysp** [comment]
It's better to separate the modification of vendor to another pr.

**Comment by MyonKeminta** [comment]
Oh, yes. It may be separated out from this PR later.

**Comment by MyonKeminta** [comment]
@tiancaiamao PTAL thanks!

File: store/tikv/gcworker/gc_worker.go [hunk 1]
```diff
@@ -356,33 +370,104 @@ func (w *GCWorker) deleteRanges(ctx context.Context, safePoint uint64) error {
...
 	metrics.GCHistogram.WithLabelValues("delete_ranges").Observe(time.Since(startTime).Seconds())
 	return nil
 }
 
+// TODO: Is there any better name for this?
+func (w *GCWorker) reDeleteRanges(ctx context.Context, safePoint uint64) error {
+	metrics.GCWorkerCounter.WithLabelValues("re_delete_range").Inc()
+
+	// We check delete range records that are deleted about 24 hours ago.
+	safePointTime := oracle.PhysicalToTime(oracle.ExtractPhysical(safePoint))
```
Comment (coocood) [inline store/tikv/gcworker/gc_worker.go]: I think calculation on the TS directly is simpler, may need to extract the constant.
```
reDeleteTs := safepoint + oracle.ComposePhysical(24 * 60 * 60 * 1000), 0)
```

**Comment by zhangjinpeng1987** [comment]
LGTM

**Pull request merged**